src/BinaryPlatforms_compat.jl

```diff
@@ -64,6 +64,8 @@ end
 
 const PlatformUnion = Union{Linux,MacOS,Windows,FreeBSD}
 
+Base.convert(::Type{Platform}, p::PlatformUnion) = p.p
```

Comment:
I'm highlighting this change.

Here we convert most of the `AbstractPlatform` subtypes to `Platform` by unwrapping them.

The one outlier is `UnknownPlatform` but there is another pull request to address that.